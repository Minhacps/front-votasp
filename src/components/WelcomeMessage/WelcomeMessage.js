import React from 'react';
import { Link } from 'react-router-dom';

import './WelcomeMessage.css';

const regions = {
  sp: { name: 'São Paulo', projectName: 'VotaSP' }
};

const WelcomeMessage = ({ regionTag, userName }) => {
  const region = regions[regionTag] || regions['sp'];

  return (
    <section>
      <div className="container welcome-text">
        <h2>
          <strong>Olá, {userName}</strong>
        </h2>
        <p>
          O projeto {region.projectName} nasceu com o objetivo de ajudar a população do estado de
          {` ${region.name}`} a escolher seus candidatos. Através de questões objetivas os eleitores
          podem verificar suas afinidades com os candidatos.
          <br />
          <strong>É muito simples e prático!</strong>
        </p>

        <Link to={`/app/questionario/1`} className="btn btn-primary btn-lg">
          Começar
        </Link>
      </div>
    </section>
  );
};

export default WelcomeMessage;
