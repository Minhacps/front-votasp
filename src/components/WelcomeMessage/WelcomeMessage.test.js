import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import WelcomeMessage from './WelcomeMessage';

describe('<WelcomeMessage />', () => {
  it('should render component without props', () => {
    const welcomeMessage = shallow(<WelcomeMessage userName="Foo Bar" />);

    expect(toJson(welcomeMessage)).toMatchSnapshot();
  });

  it('should accept valid regionTag', () => {
    const welcomeMessage = shallow(<WelcomeMessage regionTag="sp" userName="Foo Bar" />);

    expect(toJson(welcomeMessage)).toMatchSnapshot();
  });

  it('should use sp if invalid regionTag', () => {
    const welcomeMessage = shallow(<WelcomeMessage regionTag="huehuehue" userName="Foo Bar" />);

    expect(toJson(welcomeMessage)).toMatchSnapshot();
  });
});
