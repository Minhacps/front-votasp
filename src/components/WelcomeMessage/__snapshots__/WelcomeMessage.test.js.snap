// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeMessage /> should accept valid regionTag 1`] = `
<section>
  <div
    className="container welcome-text"
  >
    <h2>
      <strong>
        Olá, 
        Foo Bar
      </strong>
    </h2>
    <p>
      O projeto 
      VotaSP
       nasceu com o objetivo de ajudar a população do estado de
       São Paulo
       a escolher seus candidatos. Através de questões objetivas os eleitores podem verificar suas afinidades com os candidatos.
      <br />
      <strong>
        É muito simples e prático!
      </strong>
    </p>
    <Link
      className="btn btn-primary btn-lg"
      replace={false}
      to="/app/questionario/1"
    >
      Começar
    </Link>
  </div>
</section>
`;

exports[`<WelcomeMessage /> should render component without props 1`] = `
<section>
  <div
    className="container welcome-text"
  >
    <h2>
      <strong>
        Olá, 
        Foo Bar
      </strong>
    </h2>
    <p>
      O projeto 
      VotaSP
       nasceu com o objetivo de ajudar a população do estado de
       São Paulo
       a escolher seus candidatos. Através de questões objetivas os eleitores podem verificar suas afinidades com os candidatos.
      <br />
      <strong>
        É muito simples e prático!
      </strong>
    </p>
    <Link
      className="btn btn-primary btn-lg"
      replace={false}
      to="/app/questionario/1"
    >
      Começar
    </Link>
  </div>
</section>
`;

exports[`<WelcomeMessage /> should use sp if invalid regionTag 1`] = `
<section>
  <div
    className="container welcome-text"
  >
    <h2>
      <strong>
        Olá, 
        Foo Bar
      </strong>
    </h2>
    <p>
      O projeto 
      VotaSP
       nasceu com o objetivo de ajudar a população do estado de
       São Paulo
       a escolher seus candidatos. Através de questões objetivas os eleitores podem verificar suas afinidades com os candidatos.
      <br />
      <strong>
        É muito simples e prático!
      </strong>
    </p>
    <Link
      className="btn btn-primary btn-lg"
      replace={false}
      to="/app/questionario/1"
    >
      Começar
    </Link>
  </div>
</section>
`;
